src/code/audio_synthesis.c

```diff
@@ -682,9 +681,8 @@ Acmd* AudioSynth_DoOneAudioUpdate(s16* aiBuf, s32 aiBufLen, Acmd* cmd, s32 updat
         i++;
     }
 
-    updateIndex = aiBufLen * 2;
-    aInterleave(cmd++, DMEM_TEMP, DMEM_LEFT_CH, DMEM_RIGHT_CH, updateIndex);
-    aSaveBuffer(cmd++, DMEM_TEMP, aiBuf, updateIndex * 2);
+    aInterleave(cmd++, DMEM_TEMP, DMEM_LEFT_CH, DMEM_RIGHT_CH, aiBufLen * (s32)SAMPLE_SIZE);
```

Comment:
I guess it's debatable if `aInterleave`'s `count` is the input size `aiBufLen * (s32)SAMPLE_SIZE`, or the output length `aiBufLen * 2` (2 being the "stereo" 2 here)

sm64 abi.h https://github.com/n64decomp/sm64/blob/1372ae1bb7cbedc03df366393188f4f05dcfc422/include/PR/abi.h#L904-L924
documents it as the former "input size `aiBufLen * (s32)SAMPLE_SIZE`"

oot audio ucode https://github.com/Thar0/oot/blob/b799db384447bf42b2f1b376a362c0e3b4234117/rsp/aspMain.s#L342-L383

```as
    addi    @@count, @@count, -8
    addi    @@dmem_out, @@dmem_out, 0x10
```
I myself see `count -= 8` is "remaining output length" and `dmem_out += 0x10` is "advance 0x10 = 8 * sample_size bytes" ?

Seeing it as the output length makes more sense to me since `aiBufLen` is closer in naming to the output than input

But it's not a big deal, just arbitrary I guess.
We just gotta keep this consistent with the above blob of assembly (if one 2 is `SAMPLE_SIZE` the other is the "stereo" 2)

Reply:
I'm actually leaning towards the "output" interpretation, where the `* 2` above is indeed "two stream" as opposed to `SAMPLE_SIZE`, because if these were 8-bit audio samples, you would still need the `* 2` there